### PR TITLE
#667 - Added tests for ProcOf class.

### PR DESCRIPTION
**Pull request by driver733**
As per #667 
I have written all necessary tests for the ProcOf class.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#700](https://github.com/yegor256/cactoos/issues/700) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#700](https://github.com/yegor256/cactoos/issues/700) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C63314D6Z/520525030). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/test/java/org/cactoos/func/ProcOfTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        final List<String> list = new ArrayList<>(1);
+        new ProcOf<String>(
+            new Runnable() {
+                @Override
+                public void run() {
+                    list.add(str);
+                }
+            }
+        ).exec("You can use any input in a Runnable.");
+        MatcherAssert.assertThat(
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/func/ProcOfTest.java]: @driver733 a line can have up to 80 chars, it's not necessary to have each param on its line: 

```java
MatcherAssert.assertThat(
    "List doesn't contain expected String!", list, Matchers.contains(str)
);
```

File: src/test/java/org/cactoos/func/ProcOfTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            "Wrong result with a Runnable.",
+            list,
+            Matchers.contains(str)
+        );
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/func/ProcOfTest.java]: @driver733 How many times will ``list.add(...)`` be called? I guess once, can we have an assertion to check that the list has only one element?

File: src/test/java/org/cactoos/func/ProcOfTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            new Callable<String>() {
+                @Override
+                public String call() throws Exception {
+                    list.add(str);
+                    return "Any return value can be used.";
+                }
+            }
+        ).exec("You can use any input in a Callable");
+        MatcherAssert.assertThat(
+            "Wrong result with a Callable.",
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/func/ProcOfTest.java]: @driver733 Here as well, a more detailed failute message and indentation :)

File: src/test/java/org/cactoos/func/ProcOfTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import org.junit.Test;
+
+/**
+ * Test case for {@link ProcOf}.
+ *
+ * @author Mikhail Yakushin (driver733@me.com)
+ * @version $Id$
+ * @since 0.3
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 line)
```
Comment (amihaiemil) [inline src/test/java/org/cactoos/func/ProcOfTest.java]: @driver733 I don't think we have any magic number here, so we can remove this escaping. Also, leave the JavadocMethodCheck to 150 lines.

File: src/test/java/org/cactoos/func/ProcOfTest.java (hunk 1 above)
Comment (driver733) [reply src/test/java/org/cactoos/func/ProcOfTest.java]: @amihaiemil I believe that it is fine the way it is. We have parameters in constructors each on a separate line. In addition, it is easier to visually process (analyze) the function.

File: src/test/java/org/cactoos/func/ProcOfTest.java (hunk 2 above)
Comment (driver733) [reply src/test/java/org/cactoos/func/ProcOfTest.java]: @amihaiemil This condition is already checked by the `Matcher.contains`
See [this](https://stackoverflow.com/a/38188509/2441104) for more info.

File: src/test/java/org/cactoos/func/ProcOfTest.java (hunk 4 above)
Comment (driver733) [reply src/test/java/org/cactoos/func/ProcOfTest.java]: @amihaiemil I have removed the MagicNumberCheck.
I think that we should keep JavadocMethodCheck for 500 lines for consistency. All our tests have it.

**Comment by driver733** [comment]
@amihaiemil I have addressed the concerns you specified.

**Comment by amihaiemil** [comment]
@driver733 ok, but I meant something else with the messages, they are still not descriptive enough. They should say exactly what didn't work. E.g. "Expected list with 1 element" the fact that it was called with Runnable or something else should not necesarily be in the Error message since it is visible in the test's name.

**Comment by driver733** [comment]
@amihaiemil Done.

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by amihaiemil** [comment]
@yegor256 this can be merged as well

**Comment by amihaiemil** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job, as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by ypshenychka** [comment]
@amihaiemil According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by amihaiemil** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@amihaiemil Thank you

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
`QA` review completed: +8 points just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka), total is [+702](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality was "acceptable": +15 points just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), total is [+1010](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
> @0crat quality acceptable ([here](https://github.com/yegor256/cactoos/issues/700#issuecomment-368482186))

@ypshenychka You can't do that, unless you have one of these roles: `ARC`, `PO`. Your current roles are: `QA`.

**Comment by 0crat** [comment]
This pull request [#700](https://github.com/yegor256/cactoos/issues/700) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol), here is [why](https://www.0crat.com/footprint/C63314D6Z/591816705). The budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4). Please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by yegor256** [comment]
@0crat refuse

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/700#issuecomment-369356282))

@yegor256 The user @neonailol[/z](https://www.0crat.com/u/neonailol) resigned from [#700](https://github.com/yegor256/cactoos/issues/700), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/700#issuecomment-369356298))

@yegor256 The job [#700](https://github.com/yegor256/cactoos/issues/700) is now out of scope

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/cactoos/issues/700#issuecomment-369356298))

@yegor256 Job [#700](https://github.com/yegor256/cactoos/issues/700) was not in scope

**Pull request merged**